### PR TITLE
Makes Plant-B-Gone available to cargo, and Diona vulnerable.

### PR DESCRIPTION
**Pull request by IProduceWidgets**
## About the PR

Allows cargo to purchase Plant-B-Gone jugs.
Adjusts blasters/pistol count in watergun crate.
Adds an admeme watergun with a huge tank and high firerate.
Reduces solution cost to fire waterguns from 10u to 5u.
Makes Diona vulerable to being shot with waterguns (and spray cans) filled with WeedKiller or PlantBGone.
Adjusts damage from water to slimes up from 2 to 4. In testing it was rather low.

## Why / Balance

Improve kudzu end-game, as well as interdepartmental interactions. Helps the station fight kudzu at the cost of some credits.

It makes sense that Diona would be susceptible to herbicides, the damage values are meant to be somewhat similar to conventional firearms when used this way since its pretty niche.

Watergun adjustments were to improve the feel of the weapons, while also trying to prevent the need for a truly appalling amount of chemical.

## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

- add: Monstano brand Plant-B-Gone Jugs are now available for purchase at cargo, spray bottles not included.
- add: Diona are now susceptible to contact with herbicides.
- tweak: Cargo's waterguns are now a little less solution thirsty.

### CODE REVIEW

File: Resources/Prototypes/Entities/Structures/Storage/Tanks/tanks.yml [hunk 1]
```diff
@@ -157,3 +157,68 @@
...
+    maxFillLevels: 7
+    fillBaseName: highwatertank-2-
+  - type: SolutionContainerManager
+    solutions:
+      tank:
+        reagents:
+        - ReagentId: WeedKiller
+#         This is *100, 000* on /tg/
+          Quantity: 10000
+        maxVol: 10000
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Structures/Storage/Tanks/tanks.yml:219]: This is enough to get 500 puddles so probably should be capped to 1 or 2 k, look at water / fuel.

Comment (IProduceWidgets) [reply Resources/Prototypes/Entities/Structures/Storage/Tanks/tanks.yml:219]: Whoops, I should have been more clear.
This PR adds the regular tank size (1500u) and the high capacity tank (10000u) for completeness.
However only the regular size tank is available to cargo.

You're looking at the High Capacity tank, its not available anywhere but the entity spawn panel.

**Comment by Partmedia** [comment]
I don't like that it shares a sprite with the water tank, and that such large quantities of weed killer are available. I think it would be better to offer these for purchase as individual jugs. Or, adding a chemistry recipe so that the chemist can produce this.

**Comment by IProduceWidgets** [comment]
Hmm. Agreed, I like jugs better conceptually. I'll switch to that.

I'm hesitant to add it to chem though because it does need quite a bit to be effective against kudzu and I wanted to avoid a situation where chem feels they need to make a whole bunch of it every shift. I'd rather it be a situational solution cargo can muster if the station is overrun. Plus cargo has access to the water guns at the same time, while chem only has a couple spray bottles at best and functionally they would be pretty similar to the weed spray cans botany has already.

**Comment by Partmedia** [comment]
Thanks.

> I wanted to avoid a situation where chem feels they need to make a whole bunch of it every shift

I don't think this is an issue; it's up to chemistry to produce what they think they'll need/be useful. It's better to give them the option to make this themselves if they really wanted to.

(p.s. I'm not asking for a recipe to be added in this PR, but if you would like to add one, please go ahead)

**Comment by IProduceWidgets** [comment]
- Switched from tanks to jugs as discussed.
- Switched to PlantBGone from WeedKiller because the only real difference is WeedKiller does Poison damage and PlantBGone does a mix of brute damages. PlantbGone is way less perma-murdery. Honesly, this exercise has sort of convinced me that WeedKiller should be less available, but at least its restricted to Botany for the most part. This also solves the problems of whether it should be in chem as PlantBGone already has a recipe.
- Adjusted cargo watergun crate to have 2 of the better blasters.
- Adjusted SolutionAmmoProviderComponent (only used in waterguns) to use 5u instead of 10u per shot.
- Made Diona susceptible to weedkiller and PlantBGone like Slimes are to water.

**Comment by IProduceWidgets** [comment]
- Switched from tanks to jugs as discussed.
- Switched to PlantBGone from WeedKiller because the only real difference is WeedKiller does Poison damage and PlantBGone does a mix of brute damages. PlantbGone is way less perma-murdery. Honestly, this exercise has convinced me that WeedKiller should be less available, but at least its restricted to Botany for the most part. This also solves the problems of whether it should be in chem as PlantBGone already has a recipe.
- Adjusted cargo watergun crate to have 2 of the better blasters.
- Adjusted SolutionAmmoProviderComponent (only used in waterguns) to use 5u instead of 10u per shot. Open to suggestions on balance here since flamethrowers are also to be considered.
- Made Diona susceptible to weedkiller and PlantBGone like Slimes are to water. (a note, sprays often hit twice for some reason, but watergun shots correctly hit once.)
- Adjusted Slime weakness to water up to 4 heat from 2 heat. It was a little low in testing.

**Comment by LankLTE** [comment]
Another diona nerf let's fucking gooo

>  Adjusted SolutionAmmoProviderComponent (only used in waterguns) to use 5u instead of 10u per shot, but increased fire rates. Open to suggestions on balance here since flamethrowers are also to be considered.
>     Made Diona susceptible to weedkiller and PlantBGone like Slimes are to water. (a note, sprays often hit twice for some reason, but watergun shots correctly hit once.)
>     Adjusted Slime weakness to water up to 4 heat from 2 heat. It was a little low in testing.

I advise not making random unrelated changes in your PR, regardless of the balancing discussion (because I don't think these changes are necessary anyways)

**Comment by Partmedia** [comment]
Yes, it is better to split up unrelated changes into a separate PR. That way, if a super water blaster or diona change ends up leading to a lot of discussion, that discussion won't block the meat of the changes.

I'll try to take a closer look at your most recent changes soon; feel free to wait until somebody takes a closer look before making changes.

**Comment by IProduceWidgets** [comment]
Thanks All. Until some discussion I think this is where it makes sense for me to stop making changes.

If you all would prefer I can break them into separate PRs for the solutions, the waterguns, and the diona/slime adjustment. It just made sense to me to keep it all under one umbrella of 'changes to make kudzu events more dynamic.'

The short list of things I'm unsure about balance-wise related to the PR are:

- Is the touch damage for Diona too extreme? I dialed it back considerably from what the solutions do inject/ingested, and this PR uses the PlantBGone which can't murderbone since it stops hitting you at 200 on mixed brute damages, but currently there is still a very hefty amount of WeedKiller available to Botany from their vendomat which even without this PR does 40 poison/5u to Diona. 

- Based on above, maybe I should remove a lot of the WeedKiller available to Botany and replace most of it with PlantBGone? Its either too niche to matter or players haven't noticed how dangerous it is. 

- Watergun spray size 10u -> 5u. I also increased the rate of fire so I think it feels okay for flamethrowers, but everyone does those differently so maybe I didn't think of something.

Thanks for comments.

File: Resources/Prototypes/Entities/Mobs/Species/diona.yml [hunk 2]
```diff
@@ -35,6 +35,44 @@
...
+    - reagents: [ PlantBGone ]
+      methods: [ Touch ]
+      effects:
+      - !type:HealthChange
+        scaled: true
+        damage:
+          types:
+            Blunt: 7
+            Slash: 5
+            Piercing: 8
```
Comment (LankLTE) [inline Resources/Prototypes/Entities/Mobs/Species/diona.yml:57]: This should not be doing anywhere near enough damage to kill in 5 shots. Waterguns shoot incredibly quickly and the cargo changes will make Plant-B-Gone incredibly available- Diona are in a bad enough spot balance wise as-is. Something akin to slime's water damage should be used.

File: Resources/Prototypes/Entities/Mobs/Species/diona.yml [hunk 3]
```diff
@@ -35,6 +35,44 @@
...
+        messages: [ "diona-hurt-by-herbicide-popup" ]
+        probability: 0.66
+    - reagents: [ WeedKiller ]
+      methods: [ Touch ]
+      effects:
+      - !type:HealthChange
+        scaled: true
+        damage:
+          types:
+            Poison: 11
```
Comment (LankLTE) [inline Resources/Prototypes/Entities/Mobs/Species/diona.yml:70]: Ditto for this, poison is tough to heal and capable of perma-killing very easily so water guns should not be doing this much.

File: Resources/Prototypes/Entities/Mobs/Species/slime.yml [hunk 4]
```diff
@@ -54,7 +54,7 @@
         scaled: true
         damage:
           types:
-            Heat: 2
+            Heat: 4
```
Comment (LankLTE) [inline Resources/Prototypes/Entities/Mobs/Species/slime.yml:57]:
```suggestion
            Heat: 2
```

This change is unnecessary, for similar reasons as above.

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Basic/watergun.yml [hunk 5]
```diff
@@ -84,7 +121,7 @@
   - type: SolutionContainerManager
     solutions:
       chamber:
-        maxVol: 100 #10 shots
+        maxVol: 2500 #500 shots
```
Comment (LankLTE) [inline Resources/Prototypes/Entities/Objects/Weapons/Guns/Basic/watergun.yml:124]: Why is this changed to be so high? Something like 500 would make more sense if its' intended to be an upgrade.

**Review by LankLTE — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/Entities/Mobs/Species/diona.yml (hunk 3 above)
Comment (IProduceWidgets) [reply Resources/Prototypes/Entities/Mobs/Species/diona.yml]: Dropped to 5 Poison (20 shot kill, 40 shot round remover).

File: Resources/Prototypes/Entities/Mobs/Species/slime.yml (hunk 4 above)
Comment (IProduceWidgets) [reply Resources/Prototypes/Entities/Mobs/Species/slime.yml]: Alright, that's fair. This one was a bit of an afterthought for the PRs purpose anyway.

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Basic/watergun.yml (hunk 5 above)
Comment (IProduceWidgets) [reply Resources/Prototypes/Entities/Objects/Weapons/Guns/Basic/watergun.yml]: Agreed.
Changed to 500u max volume (100 shots).

File: Resources/Prototypes/Entities/Mobs/Species/diona.yml (hunk 2 above)
Comment (IProduceWidgets) [reply Resources/Prototypes/Entities/Mobs/Species/diona.yml]: That's fair.
Dropped to 7 overall, (15 shot crit, cannot full kill)
- 2 Blunt
- 2 Slash
- 3 Piercing

**Review by LankLTE — APPROVED** [review]

**Comment by Partmedia** [comment]
Starting the client after merging this results in:

```
[WARN] loc: "/Locale/en-US/prototypes/catalog/fills/crates/botany-crates.ftl":
"
 1 |ent-CrateHydroponicsSeedsExotic = Exotic seeds crate
    .desc = Any entrepreneuring botanist's dream. Contains many different exotic seeds. Requires Hydroponics access to open.

ent-CrateHydroponicsSeedsMedicinal = Medicinal seeds crate
    .desc = The wannabe chemist's dream. The power of medicine is at your fingertips! Requires Hydroponics access to open.

ent-CrateHydroponicsTools = Hydroponics equipment crate
    .desc = Supplies for growing a great garden! Contains some spray bottles of plant chemicals, a hatchet, a mini-hoe, scythe, as well as a pair of leather gloves and a botanist's apron.

ent-CrateHydroponicsSeeds = Seeds crate
    .desc = Big things have small beginnings. Contains twelve different seeds.

ent-CratePlantBGone = Bulk Plant-B-Gone crate
    .desc = From Monstano. \"Unwanted Weeds, Meet Your Celestial Roundup!\"
        
    ^ Expected one of \"a-zA-Z\""
```

Was this tested before merging?
